File: hw/tipantiza/u3/hw27ObserverPattern/ObserverPattern_js/model/IBM.js
const Stock = require('./Stock');

class IBM extends Stock {
    constructor(symbol, price) {
        super();
        this._symbol = symbol;
        this._price = price;
    }

    setPrice(price) {
        this._price = price;
        this.notifyObservers(price);
    }

    setSymbol(symbol) {
        this._symbol = symbol;
        this.notifyObservers(symbol);
    }
}

module.exports = IBM;

File: hw/tipantiza/u3/hw27ObserverPattern/ObserverPattern_js/ObserverPattern/src/main/java/ec/edu/espe/observerpattern/model/IBM.java

package ec.edu.espe.observerpattern.model;

/**
 *
 * @author Alexander Tipantiza, The Softwarrios, @ESPE
 */

public class IBM extends Stock {
    public IBM(String symbol, double price) {
        this.symbol = symbol;
        this.price = price;
    }

    public void setPrice(double price) {
        this.price = price;
        notifyObservers(price);
    }

    public void setSymbol(String symbol) {
        this.symbol = symbol;
        notifyObservers(symbol);
    }
}